### PR TITLE
Thrones, chests, and stack crafting changes

### PR DESCRIPTION
**Pull request by MadmanMartian**
Adds thrones. Made from 40 wood planks, takes 10 seconds to build, and will make the captain green with envy.

Also removes copypaste from the chair section, as it was getting weird in there.

Sprite and idea courtesy of @Blithering 

thrones

Adds chests to the wood crafting section, requiring 10 wood and 5 plasteel. The plasteel either needs to be in your vicinity or in your off-hand. A bit of a PoC of stack recipes requiring something more than just the initial stack associated with them.

:cl:
* rscadd: Can now build thrones from wood.
* rscadd: Can now build chests from wood and plasteel. The plasteel has to be in your immediate vicinity or in your off-hand.

### CODE REVIEW

**Comment by DeityLink** [comment]
Reminds me that stone still isn't a material.

**Comment by Diavill** [comment]
where sprite

**Comment by ultraplen** [comment]
Been a while since I worked on the chair code, but have you made sure that all the chair layers work fine after your copy-paste removal? I think I recall that bunch of the chairs need snowflake stuff to make the layers appear correctly.

**Comment by MadmanMartian** [comment]
Their specific snowflake bits should still work, I just moved the overlay adding and removing down to regular chair level as that was the bit that was redundantly copypasted

File: code/game/objects/structures/stool_bed_chair_nest/chairs.dm [hunk 1]
```diff
@@ -18,6 +18,25 @@
 	..()
 	handle_layer()
 
+/obj/structure/bed/chair/lock_atom(var/atom/movable/AM)
+	. = ..()
+	update_icon()
+
+/obj/structure/bed/chair/unlock_atom(var/atom/movable/AM)
+	..()
```
Comment (DamianX) [inline code/game/objects/structures/stool_bed_chair_nest/chairs.dm]: This is going to completely turbofuck vehicles and possibly others, make sure the return value is passed along

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -93,7 +93,33 @@
 				title+= "[R.title]"
 			//title+= " ([R.req_amount] [src.singular_name]\s)"
 			title+= " ([R.req_amount] [CORRECT_STACK_NAME(src)]"
-
+			if(R.other_reqs.len)
+				for(var/ii=1;ii<=R.other_reqs.len,ii++)
```
Comment (PJB3005) [inline code/game/objects/items/stacks/stack.dm]: `for (var/ii = 1 to R.other_reqs.len)`

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -168,6 +194,22 @@
 		//	//new_item.add_to_stacks(usr)
 
 		src.use(R.req_amount*multiplier)
+		if(R.other_reqs.len)
```
Comment (PJB3005) [inline code/game/objects/items/stacks/stack.dm]: This is gonna break. You can still build it provided the UI doesn't update.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Review by PJB3005 — CHANGES_REQUESTED** [review]
Now the building code uses materials even if the construction fails to find all materials.

**Comment by MadmanMartian** [comment]
@PJB3005 changes made

**Review by jknpj — APPROVED** [review]

**Pull request merged**